A8C Libs S3 and Fetch Style Plugin

ParaskP7 opened the pull request:
### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR is twofold:

1. It adds the missing `A8C Libs S3` settings related repositories configuration. (2c381c68bf8da41e3f273c6ce193243b5404d264)
2. It removes the [fetchstyle](https://github.com/Automattic/style-config-android) plugin from the project. (cb57d7c9a7299cccc197a2a65b77e4f03fde3510)

PS: For more info please refer to this internal discussion here: `CC7L49W13/p1667892065193709`

-----

This [style-config-android](https://github.com/Automattic/style-config-android) plugin is mostly unused and very outdated.

To the @woocommerce/apps-infrastructure's knowledge and checking GE stats no-one is using the `./gradlew downloadConfigs` task to fetch/update the style config files. For years now, almost all Android engineers depend on the default AS style and such like `.idea` configuration.

In addition to that, this plugin has recently created a build related issue where it couldn't be found in any of the configured sources (see [here](https://buildkite.com/automattic/woocommerce-android/builds/7477) and [here](https://buildkite.com/automattic/woocommerce-android/builds/7478#018456c6-4b17-4413-b2f0-c5c80b366622)). Keeping and maintaining such an unused/outdated plugin is counter productive.

```
* What went wrong:
[2022-11-08T07:04:47Z] Plugin [id: 'com.automattic.android.fetchstyle',
artifact: 'com.automattic.android:fetchstyle:1.1'] was not found in any
of the following sources:
```

As such, this [plugin](https://github.com/Automattic/style-config-android) is now removed and will be (most probably) also removed on all other repos (sooner or later). Then, the [plugin's repo](https://github.com/Automattic/style-config-android) will be then marked as deprecated and archived.

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough.
- However, if you want to be thorough about reviewing this change, you could try and run the `./gradlew downloadConfigs` task and verify that it is no longer found in the root project for `WCAndroid`.

Note that the [dependency-tree-diff](https://buildkite.com/automattic/woocommerce-android/builds/7481#018456e0-d322-4c02-977d-e9482e785a99) CI check that is failing is expected since this `fetchstyle` and its configuration is still available on `trunk` and as such the dependency tree diff cannot be completed as it also failing on that branch (as expected).

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

ParaskP7 committed:
Build: Temporary settings repositories commit to trigger ci

ParaskP7 committed:
Build: Add missing settings a8c libs s3 repositories configuration

ParaskP7 committed:
Build: Remove fetchstyle plugin from project

This 'style-config-android' plugin is mostly unused and very outdated.

To the Apps Infra's knowledge and checking GE stats no-one is using the
'./gradlew downloadConfigs' task to fetch/update the style config files.
For years now, almost all Android engineers depend on the default AS
style and such like '.idea' configuration.

------------------------------------------------------------------------

In addition to that, this plugin has recently created an build related
issue where it couldn't be found in any of the configured sources (see
below). Keeping and maintaining such an unused/outdated plugin is
counter productive.

"* What went wrong:
 [2022-11-08T07:04:47Z] Plugin [id: 'com.automattic.android.fetchstyle',
 artifact: 'com.automattic.android:fetchstyle:1.1'] was not found in any
 of the following sources:"

------------------------------------------------------------------------

As such, this plugin is now removed and will be (most probably) also
removed on all other repos (sooner or later). Then, the plugin's repo
will be then marked as deprecated and archived.

Plugin Link: https://github.com/Automattic/style-config-android

oguzkocer commented on settings.gradle:
```diff
@@ -6,6 +6,18 @@ pluginManagement {
+            }
+            filter {
+                includeGroup "com.automattic.android"
+                includeGroup "com.automattic.android.configure"
+                includeGroup "com.automattic.android.publish-to-s3"
```
We shouldn't need either of these, because as far as I know we are not using the `configure` plugin in this repo and `publish-to-s3` plugin is definitely not used since this is a client and not a library. I might be mistaken, but I think we don't need this block all together.

ParaskP7 replied:
Ah, great thanks for the clarification on that @oguzkocer ! 🙇 

I keep being confused with this `configure` plugin and its `updateConfiguration` and `applyConfiguration` tasks, that is, versus the `configureApply` task on both the WPAndroid and WCAndroid. Can this be something to align on `FluxC` side as well, that is, to have that `com.automattic.android.configure` be removed there too and replace that with `configureApply`, just like it is done on these other 2 clients repos? 🤔

ParaskP7 replied:
Btw, do we maybe need to keep the `includeGroup "com.automattic.android"` configuration @oguzkocer ? 🤔

I am seeing on WPAndroid we have the below settings related repositories configuration:

```
        maven {
            url 'https://a8c-libs.s3.amazonaws.com/android'
            content {
                includeGroup "com.automattic.android"
            }
        }
```

ParaskP7 replied:
This is now done here: 8ff4522b9bea88466efe94be366d7d15d96b11eb

PS: Let me know if you think we should remove whole block anyway.

wzieba replied:
> I am seeing on WPAndroid we have the below settings related repositories configuration:

I think they have this because they do use `fetchstyle` plugin ([source](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/settings.gradle#L21))

ParaskP7 committed:
Build: Remove unnecessary configure and publish-to-s3 repo configuration

For more info see PR comment here: https://github.com/woocommerce/
woocommerce-android/pull/7755#discussion_r1016463239

ParaskP7 commented:
👋 @woocommerce/android-developers !

I am pinging you all here, also adding you all as reviewers, so that you are aware of this change coming-in.

In case you still, for some reason, prefer to keep on hold and use this [fetchstyle](https://github.com/Automattic/style-config-android) plugin, let us know before we complete remove this from this and all other repos. 🙏 

PS: Apologies for making this decision as a last minute change but CI was failing and we took the opportunity to clean this up, seeing that this plugin is actually unused and outdated.

wzieba commented on settings.gradle:
```diff
@@ -6,6 +6,16 @@ pluginManagement {
     gradle.ext.navigationVersion = '2.4.1'
 
     repositories {
+        exclusiveContent {
+            forRepository {
+                maven {
+                    url = uri("https://a8c-libs.s3.amazonaws.com/android")
+                }
+            }
+            filter {
+                includeGroup "com.automattic.android"
+            }
+        }
```
As we don't have any `com.automattic.android.*` plugins declared, we shouldn't need to declare this repository.

```suggestion

```

ParaskP7 replied:
👋 @wzieba !

Thank you for the review, I would definitely apply your commit suggestion, but for now, how about keeping this configuration just as a default configuration so that we are ready for `com.automattic.android.*` plugins to be added and not forget to be doing so when needed in the future. It doesn't hurt, does it? 🤔 

Also, I would have applied the commit suggestion but this will force the build to start over, maybe we can unblock `trunk` now and then discuss about removing this extra (unneeded for now) repositories configuration. Wdyt? 🤔

wzieba replied:
Sure, shouldn't hurt performance-wise. I'd remove this as not-needed, but that's not a blocker. :shipit:

ParaskP7 replied:
Great, thank you @wzieba ! 🙇

Yea, I know, but now we can remove this at our pace and whenever, nothing is blocking anymore, feel free to do so as you wish and whenever if you feel strongly about having such configuration without necessarily explicitly affecting the build outcome! 💯

wzieba reviewed (approved):
Thanks @ParaskP7 ! Leaving one suggestion. It'd be awesome if we could merge this PR soon to unblock `trunk` 🙏 .
